src/headset/fake.c

```diff
@@ -0,0 +1,493 @@
+  *x = v[0];
+  *y = v[1];
+  *z = v[2];
+#endif
+// TODO
+}
+
+
+
+
+static vec_controller_t* fakeGetControllers() {
+  return &state.controllers;
+}
+
+static int fakeControllerIsPresent(Controller* controller) {
+    return 1;
+}
+
+static ControllerHand fakeControllerGetHand(Controller* controller) {
+  return HAND_UNKNOWN;
+}
+
+static void fakeControllerGetPosition(Controller* controller, float* x, float* y, float* z) {
+  // for now, locked to headset
+  fakeGetPosition(x,y,z);
+}
+
+static void fakeControllerGetOrientation(Controller* controller, float* angle, float* x, float* y, float* z) {
+  // for now, locked to headset
+  float q[4];
+  quat_fromMat4(q, state.transform);
+  quat_getAngleAxis(q, angle, x, y, z);
+}
+
+
+static float fakeControllerGetAxis(Controller* controller, ControllerAxis axis) {
+  return 0.0f;
+}
+
+static int fakeControllerIsDown(Controller* controller, ControllerButton button) {
+  GLFWwindow* window = lovrGraphicsGetWindow();
+  if(!window) {
+    return 0;
+  }
+  int b = glfwGetMouseButton( window, GLFW_MOUSE_BUTTON_LEFT);
+  return (b == GLFW_PRESS) ? CONTROLLER_BUTTON_TRIGGER : 0;
+}
+
+static int fakeControllerIsTouched(Controller* controller, ControllerButton button) {
+  return 0;
+}
+
+static void fakeControllerVibrate(Controller* controller, float duration, float power) {
+}
+
+static ModelData* fakeControllerNewModelData(Controller* controller) {
+  return NULL;
+}
+
+
+static void fakeRenderTo(headsetRenderCallback callback, void* userdata) {
+//  float head[16], transform[16], projection[16];
+
+  // TODO: Head transform
+  // TODO: Eye transform
+  // Projection
+ 
+  int w,h; 
+  GLFWwindow* window = lovrGraphicsGetWindow();
+  if(!window) {
+    return;
+  }
+
+  glfwGetWindowSize(window, &w, &h); 
+
+  float transform[16];
+  mat4_perspective(state.projection, state.clipNear, state.clipFar, 67 * M_PI / 180.0, (float)w/h);
+
+  // render
+  lovrGraphicsPush();
+  float inv[16];
+  mat4_set(inv,state.transform);
+  mat4_invert(inv);
+  lovrGraphicsMatrixTransform(MATRIX_VIEW, inv);
+
+  lovrGraphicsSetProjection(state.projection);
+  lovrGraphicsClear(1, 1);
+  callback(EYE_LEFT, userdata);
+  lovrGraphicsPop();
+}
+
+
+static void fakeUpdate(float dt)
+{
+  float k = 4.0f;
+  check_window_existance();
+  GLFWwindow* w = lovrGraphicsGetWindow();
+  if(!w) {
+    return;
+  }
+  float v[3] = {0.0f,0.0f,0.0f};
+  if (glfwGetKey(w, GLFW_KEY_W)==GLFW_PRESS || glfwGetKey(w, GLFW_KEY_UP)==GLFW_PRESS) {
+    v[2] = -k;
+  }
+  if (glfwGetKey(w, GLFW_KEY_S)==GLFW_PRESS || glfwGetKey(w, GLFW_KEY_DOWN)==GLFW_PRESS) {
+    v[2] = k;
+  }
+  if (glfwGetKey(w, GLFW_KEY_A)==GLFW_PRESS || glfwGetKey(w, GLFW_KEY_LEFT)==GLFW_PRESS) {
+    v[0] = -k;
+  }
+  if (glfwGetKey(w, GLFW_KEY_D)==GLFW_PRESS || glfwGetKey(w, GLFW_KEY_RIGHT)==GLFW_PRESS) {
+    v[0] = k;
+  }
+  if (glfwGetKey(w, GLFW_KEY_SPACE) == GLFW_PRESS) {
+    v[1] = k;
+  }
+  if (glfwGetKey(w, GLFW_KEY_LEFT_SHIFT) == GLFW_PRESS) {
```

Comment:
I wonder if this will be annoying in windows due to all the weird sticky keys/filter keys stuff.  If so, maybe it could be switched to Q/E or something.